### PR TITLE
Handle proposal list pagination and filtering

### PR DESCRIPTION
**Pull request by hochiw**
## Features
- Introduced react-paginate
- Implemented PageController component
- Implemented FilterTabs component
- Integrated both components to ProposalScreen
- Removed search bar placeholder

## Screenshots
| Desktop | Mobile |
|---------|--------|
|Screenshot 2022-06-13 at 6 23 41 PM|Screenshot 2022-06-13 at 6 24 06 PM|

based on #99 
refs #69

### CODE REVIEW

**Comment by chihimng** [comment]
@hochiw seems the gap b/t rows are different from design?

File: react-app/src/components/ProposalScreen/ProposalScreen.tsx [hunk 1]
```diff
@@ -9,18 +9,19 @@ import {
...
 } from "../../models/RequestState";
+import PageContoller from "../common/PageController/PageController";
 import { useProposalsQuery } from "./ProposalScreenAPI";
 import { ProposalList } from "./ProposalList";
 
 const PROPOSAL_LIST_PAGE_SIZE = 5;
 
 const ProposalScreen: React.FC = () => {
   const { requestState, fetch } = useProposalsQuery(0, PROPOSAL_LIST_PAGE_SIZE);
+  const [offset, setOffset] = useState<number>(0);
```
Comment (chihimng) [inline react-app/src/components/ProposalScreen/ProposalScreen.tsx:45]: better keep page number at url query param?

**Review by chihimng — CHANGES_REQUESTED** [review]
nice interfaces 👍

File: react-app/src/components/ProposalScreen/ProposalScreen.tsx [hunk 2]
```diff
@@ -76,6 +92,14 @@ const ProposalScreen: React.FC = () => {
     fetch(offset);
   }, [fetch, offset]);
 
+  useEffect(() => {
+    const page = offset / PROPOSAL_LIST_PAGE_SIZE + 1;
+    setSearchParams({
```
Comment (chihimng) [inline react-app/src/components/ProposalScreen/ProposalScreen.tsx:97]: @hochiw will this populate browser history tho? what if user press back btn?

Comment (hochiw) [reply react-app/src/components/ProposalScreen/ProposalScreen.tsx]: Added `replace: true` to prevent param reverting

Comment (chihimng) [reply react-app/src/components/ProposalScreen/ProposalScreen.tsx]: oh... i mean populating history is what we want? so user can press forward / backward as expected

Comment (hochiw) [reply react-app/src/components/ProposalScreen/ProposalScreen.tsx]: oh ok

File: react-app/src/components/ProposalScreen/ProposalScreen.tsx [hunk 3]
```diff
@@ -56,25 +68,33 @@ const ProposalScreen: React.FC = () => {
...
-        applyFilter(filter, wallet.account.address);
-        return;
-      }
-
-      applyFilter(filter, null);
-      setOffset(0);
+      const params = searchParams;
+      params.set("filter", filter);
+      setSearchParams(params);
+      setPage(0);
```
Comment (chihimng) [inline react-app/src/components/ProposalScreen/ProposalScreen.tsx:86]: can just `params.set("page", 1)` above?

**Review by chihimng — CHANGES_REQUESTED** [review]
plz rebase 🙏

**Comment by hochiw** [comment]
@chihimng updated 🙏

**Pull request merged**